pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -94,6 +98,37 @@ func (hs *proxierHealthServer) QueuedUpdate() {
 	hs.oldestPendingQueued.CompareAndSwap(zeroTime, hs.clock.Now())
 }
 
+// IsHealthy returns the proxier's health state, following the same definition
+// the HTTP server defines.
+func (hs *proxierHealthServer) IsHealthy() bool {
+	isHealthy, _, _ := hs.isHealthy()
+	return isHealthy
+}
+
+func (hs *proxierHealthServer) isHealthy() (bool, time.Time, time.Time) {
+	var oldestPendingQueued, lastUpdated time.Time
+	if val := hs.oldestPendingQueued.Load(); val != nil {
+		oldestPendingQueued = val.(time.Time)
+	}
+	if val := hs.lastUpdated.Load(); val != nil {
+		lastUpdated = val.(time.Time)
```

Comment:
@danwinship why do we need lastUpdated, it doesn't seem to be used anywhere?

Reply:
Maybe: https://github.com/kubernetes/kubernetes/blob/f286545c48fa04f609cb923ac39b5c7679098054/pkg/proxy/healthcheck/proxier_health.go#L168-L173

?

Reply:
yeah, it is always set to currentTime , no? why do we need to get this value, is not used anywhere AFAIK

Reply:
I mean, we can return `oldestPendingQueued` instead of `lastUpdated`

Reply:
nevermind, it doesn't solve anything, I just wanted to see if we can save a variable and I said nonsense, please resolve this comment